zeppelin-server/src/main/java/org/apache/zeppelin/rest/ClusterRestApi.java

```diff
@@ -0,0 +1,275 @@
+      }
+      if (properties.containsKey(ClusterMeta.INTP_PROCESS_LIST)) {
+        sortProperties.put(ClusterMeta.INTP_PROCESS_LIST, properties.get(ClusterMeta.INTP_PROCESS_LIST));
+      }
+
+      HashMap<String, Object> node = new HashMap<String, Object>();
+      node.put(ClusterMeta.NODE_NAME, nodeName);
+      node.put(PROPERTIES, sortProperties);
+
+      nodes.add(node);
+    }
+
+    return new JsonResponse(Response.Status.OK, "", nodes).build();
+  }
+
+  private String formatIntpLink(String intpName) {
+    return String.format("<a href=\"/#/cluster/%s\">%s</a>", intpName, intpName);
+  }
+
+  /**
+   * get node info by id
+   */
+  @GET
+  @Path("/node/{nodeName}/{intpName}")
+  @ZeppelinApi
+  public Response getClusterNode(@PathParam("nodeName") String nodeName,
+                                 @PathParam("intpName") String intpName){
+    ArrayList<HashMap<String, Object>> intpProcesses = new ArrayList<HashMap<String, Object>>();
+
+    Map<String, HashMap<String, Object>> intpMeta = null;
+    if (isTest) {
+      intpMeta = mockIntpMeta();
+    } else {
+      intpMeta = clusterManagerServer.getClusterMeta(ClusterMetaType.INTP_PROCESS_META, "");
+    }
+
+    // Number of calculation processes
+    for (Map.Entry<String, HashMap<String, Object>> intpMetaEntity : intpMeta.entrySet()) {
+      String intpNodeName = (String) intpMetaEntity.getValue().get(ClusterMeta.NODE_NAME);
+
+      if (null != intpNodeName && intpNodeName.equals(nodeName)) {
+        HashMap<String, Object> node = new HashMap<String, Object>();
+        node.put(ClusterMeta.NODE_NAME, intpNodeName);
+        node.put(PROPERTIES, intpMetaEntity.getValue());
+        intpProcesses.add(node);
+      }
+    }
+
+    return new JsonResponse(Response.Status.OK, "", intpProcesses).build();
+  }
+
+
+  private Map<String, HashMap<String, Object>> mockNodesMeta() {
+    Map<String, HashMap<String, Object>> serverMeta = new HashMap<>();
+
+    for (Integer i = 0; i < 32; i ++) {
+      HashMap<String, Object> nodeMeta = new HashMap<>();
+      nodeMeta.put(ClusterMeta.NODE_NAME, "127.0.0." + i.toString() + ":" + "8080");
+      nodeMeta.put(ClusterMeta.SERVER_HOST, "127.0.0." + i.toString());
+      nodeMeta.put(ClusterMeta.SERVER_PORT, "8080");
+      nodeMeta.put(ClusterMeta.MEMORY_USED, 100);
+      nodeMeta.put(ClusterMeta.MEMORY_CAPACITY, 4800);
+      nodeMeta.put(ClusterMeta.CPU_USED, 200);
+      nodeMeta.put(ClusterMeta.CPU_CAPACITY, 10000);
+      nodeMeta.put(ClusterMeta.SERVER_HOST, "host1");
+      nodeMeta.put(ClusterMeta.SERVER_PORT, 1084);
+      nodeMeta.put(ClusterMeta.STATUS, ClusterMeta.ONLINE_STATUS);
+      nodeMeta.put(ClusterMeta.SERVER_START_TIME, LocalDateTime.now());
+      nodeMeta.put(ClusterMeta.LATEST_HEARTBEAT, LocalDateTime.now());
+
+      serverMeta.put("127.0.0." + i.toString() + ":" + "8080", nodeMeta);
+    }
+
+    return serverMeta;
+  }
+
+  private Map<String, HashMap<String, Object>> mockIntpMeta() {
```

Comment:
test code should not be put in this class

Reply:
Ok, i removed.